src/stylus/components/_buttons.styl

```diff
@@ -142,24 +142,27 @@ theme(button, "btn")
     &:after
       border-radius: 50%
 
-    .icon
-      &:not(:only-of-type):last-of-type
-        opacity: 0
-        position: absolute
-        transform: rotate(-45deg)
-        left: calc(50% - 12px)
-        top: calc(50% - 12px)
+    .btn__content
+      :not(:only-child)
+        transition: $primary-transition
+
+        &:last-child
+          opacity: 0
+          position: absolute
+          transform: rotate(-45deg)
+          left: calc(50% - 12px)
+          top: calc(50% - 12px)
 
     &.btn--active
-      .icon
-        &:not(:only-of-type)
-          &:first-of-type
+      .btn__content
+        :not(:only-child)
+          &:first-child
             opacity: 0
             transform: rotate(45deg)
             left: 'calc(50% - %s)' % (24 / 2)
             top: 'calc(50% - %s)' % (24 / 2)
```

Comment:
Also, this generates invalid CSS (`calc(50% - 12)` instead of `calc(50% - 12px)`)